Shared/Samples/Measure distance in scene/MeasureDistanceInSceneView.swift

```diff
@@ -0,0 +1,175 @@
+// Copyright 2023 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+import ArcGIS
+
+struct MeasureDistanceInSceneView: View {
+    /// The view model for the sample.
+    @StateObject private var model = Model()
+    
+    /// A string for the direct distance of the location distance measurement.
+    @State private var directDistanceText = "--"
+    
+    /// A string for the horizontal distance of the location distance measurement.
+    @State private var horizontalDistanceText = "--"
+    
+    /// A string for the vertical distance of the location distance measurement.
+    @State private var verticalDistanceText = "--"
+    
+    /// The unit system for the location distance measurement, selected by the picker.
+    @State private var unitSystemSelection: UnitSystem = .metric
+    
+    /// The overlay instruction message text.
+    @State private var instructionText: String = .startMessage
+    
+    var body: some View {
+        VStack {
+            SceneViewReader { sceneViewProxy in
+                SceneView(scene: model.scene, analysisOverlays: [model.analysisOverlay])
+                    .onSingleTapGesture { screenPoint, _ in
+                        // Set the start and end locations when the screen is tapped.
+                        Task {
```

Comment:
> Can you let me know if you think there is a better way to use the task blocks

I tried a few ideas but since this sample supports both drag and single tap gestures, I can't find a good way to differentiate one from another without adding a few states. 

Thus I think in-situ task blocks are good.